Source/Shaders/GlobeVS.glsl

```diff
@@ -175,9 +176,18 @@ void main()
 #endif
 
 #ifdef APPLY_MATERIAL
-    vec3 finalNormal = normalMC;
-    vec3 ellipsoidNormal = normalize(position3DWC.xyz);
-    v_slope = abs(dot(ellipsoidNormal, finalNormal));
+    float northPoleZ = czm_getWgs84EllipsoidEC().radii.z;
+    vec3 northPolePositionMC = vec3(0.0, 0.0, northPoleZ);
+    vec3 ellipsoidNormal = normalize(v_positionMC); // TODO: For a sphere this is correct, but not generally for an ellipsoid!
+    vec3 vectorEastMC = normalize(cross(northPolePositionMC - v_positionMC, ellipsoidNormal));
+    float dotProd = abs(dot(ellipsoidNormal, v_normalMC));      // Why absolute value here?  What about overhanging terrain?  (e.g., an arch.)
```

Comment:
Since the terrain system is 2.5D (one height value per x,y coordinate) I don't think the comment necessarily applies.

Reply:
okay @lilleyse, i deleted the comment